src/Migrator.sol

```diff
@@ -215,30 +95,32 @@ contract Migrator is IMigrator, Ownable {
             fromInternalBalance: false
         });
 
-        balancerVault.joinPool(balancerPoolId, address(this), address(this), request);
-    }
+        IVault(_addresses.balancerVault).joinPool(balancerPoolId, address(this), address(this), request);
 
-    /**
-     * @notice Deposit BPT into rewards pool
-     */
-    function _depositIntoRewardsPool() internal {
-        uint256 amount = balancerPoolToken.balanceOf(address(this));
+        uint256 amountReceived = ERC20(_addresses.balancerPoolToken).balanceOf(address(this));
 
-        auraPool.deposit(amount, msg.sender);
+        // msg.sender receives auraBPT or BPT depending on their choice to deposit into Aura pool
+        if (_details.stakeBpt) {
+            IRewardPool4626(_addresses.auraPool).deposit(amountReceived, msg.sender);
```

Comment:
It might be wise to add slippage control here. I'm not sure how `IRewardPool4626` is written but there is the potential that ERC-4626 initial vault deposits/mints get sandwiched.

Reply:
We could call this function from the UI and then pass the result as a param having it act as a min amount out check.  

`    /**
     * @notice Allows an on-chain or off-chain user to simulate
     * the effects of their deposit at the current block, given
     * current on-chain conditions.
     */    
    function previewDeposit(uint256 assets) external view virtual override returns(uint256){
        return convertToShares(assets);
    }`

Reply:
added a check around minAuraBptAmountOut